### PR TITLE
NEO-1253 // resolve Checkbox Group a11y issue

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Checkbox Group](https://deploy-preview-77--neo-react-library-storybook.netlify.app/?path=/story/components-checkbox-group)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Yang discovered an issue with how we were linking the checkbox group with a given label. Specifically that we were not appropriately linking the "label" with the group of checkboxes. With this update, we are. Wether the label is passed via the `label` prop, or is linked remotely via the `aria-labelledby` prop. I checked both with VoiceOver and they acted the same.

Also, code coverage is back up to 100% for Checkbox components! 🥳 
Screen Shot 2022-11-17 at 10 39 28 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "test:cypress-open": "cypress open --component -b chrome",
     "github": "yarn && yarn lint && yarn build && yarn test && yarn test:cypress",
     "netlify": "yarn && yarn build && yarn build:storybook",
-    "all": "yarn && yarn clean && yarn format && yarn lint && yarn build && yarn jsdocs && yarn build:storybook && yarn test:cypress && yarn test:badges && yarn pack"
+    "all": "yarn && yarn clean && yarn format && yarn lint && yarn build && yarn build:storybook && yarn test:cypress && yarn test:badges && yarn pack"
```
Comment (joe-s-avaya) [inline package.json:55]: removed `yarn jsdocs &&` due to it bugging out, but we don't use it, so "meh", might fix it later

**Review by yangAtSpoken — APPROVED** [review]

File: src/components/CheckboxGroup/CheckboxGroup.stories.tsx [hunk 2]
```diff
@@ -16,12 +18,28 @@ export default {
...
     </CheckboxGroup>
   );
 };
 
+export const UseOfSeperateLabel = () => {
+  return (
+    <>
+      <div id="checkbox-group-label">Seperate Label for Group</div>
+      <CheckboxGroup
+        groupName="single-checkbox"
```
Comment (mrazian85) [inline src/components/CheckboxGroup/CheckboxGroup.stories.tsx:33]: Should this be "single-checkbox" if there are two?

Comment (joe-s-avaya) [reply src/components/CheckboxGroup/CheckboxGroup.stories.tsx:33]: ah, missed that, good catch, will update

**Review by enrique-prado — APPROVED** [review]
Looks good.

**Review by mrazian85 — APPROVED** [review]

**Pull request merged**